### ISSUE TITLE
Serial port MIDI

### ISSUE THREAD

**Issue by gridsleep** [AppleWin/AppleWin#780]
How about USB MIDI support? If AppleWin could RX through an external MIDI interface, it would save me having to purchase an actual Apple IIgs. If it were possible, would it be limited only to MIDI devices that were available to Apple II, or would any modern MIDI interface work? Considering that MIDI is standard, as long as the device appears (like a joystick would) then it should work.

**Comment by sicklittlemonkey** [comment]
There's a MIDI API in Windows, but I know nothing about it.

Presumably AppleWin would have to emulate this card (or similar) which
seems to be the standard:
https://mirrors.apple2.org.za/Apple%20II%20Documentation%20Project/Interface%20Cards/Audio/Passport%20Designs%20MIDI%20Interface%20Card/


Not an Epic task, but still a large one.

Cheers,
Nick.

**Comment by gridsleep** [comment]
The Apple II serial port can plug directly into an eMagic AMT|8 or older MOTU MIDI interface using the 8-pin DIN cable. The question is, can the 8-pin port be replicated to link through to the USB port of the host? An actual Apple II can connect with those older MIDI adapters. New MIDI adapters use USB 3 or USB C. If AppleWin has the 8-pin serial emulation, can that be made to detect things other than joysticks? The software I want to use runs perfectly in AppleWin. It needs to have access to the Apple serial port. It doesn't have to be a Passport MIDI card. MIDI hardware I have in buckets. I just need to know if the software can create and make use of a virtual Apple serial port that maps to a USB port. The only other alternative is to buy a complete Apple II system for not inconsiderable money. I would like to avoid that if a software solution exists or can be brought about.

**Comment by sicklittlemonkey** [comment]
I think Tom recently improved AppleWin to be able to use any serial device
that is listed as a COM port. See here:
https://github.com/AppleWin/AppleWin/issues/435

**Comment by tomcw** [comment]
@gridsleep - so any luck in trying to hook AppleWin's serial card support up to a MIDI device?

**Comment by gridsleep** [comment]
Applewin does not appear to be able to communicate through to the USB in
any useful way. I can use my Windows software to do exactly what I want,
but Applewin doesn't have such throughput set up. I am going to look over
the source code to see if it can be altered to give the software it is
running full access. I guess I'm going to be cross referencing a lot of
hardware and software manuals before I'm through.

**Comment by gridsleep** [comment]
I've decided that rather than try to rewrite/improve Applewin, or invest
(possibly heavily) in a physical Apple IIgs to run one program
intermittently, I will start from scratch and recreate Kurzweil's Sound
Modeling Program in Python so I can program my K150 synthesizer directly
from my Windows 10 laptop. I had to get two other computers out of physical
necessity; a PowerMac G4 (only $83 total) for my Korg OASYS-PCI card, and a
32-bit Dell laptop ($24) for my Hartmann Neuron VS. Apple IIgs are going
for ridiculously higher prices for no accountable reason. Although the IIgs
has the nice Ensoniq DOC audio chip, that kind of sound is much more
convenient in an actual Ensoniq keyboard. I haven't written a challenging
piece of software in a very long time and pandemic isolation is the perfect
environment. Cheers.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by tomcw** [comment]
As Nick says above:
>Presumably AppleWin would have to emulate this card (Passport MIDI)

This has an MC6850 (ACIA: 2x read & 2x write I/O registers) and an MC6850 (Programmable Timer Module: 8x I/O registers - has 3x 16 bit timers). We'd need to map the ACIA (serial) to either a real serial or USB MIDI interface on the PC.

>Not an Epic task, but still a large one.

Yep. Time and motivation are other factors!

Also to test the implementation would require a (USB) MIDI adapter (eg. this [Gelrhonr USB MIDI cable adapter](https://www.amazon.co.uk/Gelrhonr-Interface-Indicator-Keyboard-Vista-6-5Ft/dp/B093L45BQD/ref=sr_1_4?keywords=gelrhonr+midi+adapter&qid=1678487023&sr=8-4) for £10) + a keyboard with MIDI in.

**Comment by ajack2001my** [comment]
If the Passport MIDI emu could link to Windows MIDI mapper, that'll be great too...

**Comment by tomcw** [comment]
There are software MIDI synths too:
- "Microsoft GS Wavetable Synth" - generally gets poor reviews and seems like this isn't available from Win10 ([ref](https://answers.microsoft.com/en-us/insider/forum/all/microsoft-gs-wavetable-synth-on-windows-10-not/7ea6269c-3c32-462c-b851-7e98568c0b3f))
- http://coolsoft.altervista.org/en/virtualmidisynth

**Comment by ajack2001my** [comment]
Yes, the two you mentioned will hook up to the MIDI Mapper and service any MIDI calls that come their way. I use both software/hardware MIDI sound modules on my setup.

**Comment by rb6502** [comment]
I think OP is a little confused here.  MIDI serial port interfaces only worked on the Apple IIgs and Mac because they relied on the Zilog 8530 serial chip and the way Apple hooked it up in those machines (you could feed in the serial clock through one of the pins to get the weird MIDI 31250 baud rate).  You can't use one with a Super Serial Card on a II Plus or IIe, and AppleWin doesn't emulate the IIgs.

As far as I know MAME is the only emulator currently supporting MIDI I/O for the Apple II (we emulate the Yamaha/Passport 6840+6850 card, which also works on the IIgs with synthLAB).

**Comment by tomcw** [comment]
re. other Apple II emulators that support Passport MIDI
There's [JACE](https://github.com/badvision/jace), which looks to have supported this since 2012!
YT video here: https://www.youtube.com/watch?v=1G3wQiKtzJ8
